### PR TITLE
Fix crash when trying to import a non-exported type

### LINKED ISSUE #36592: Crash: Cannot read property 'kind' of undefined
Repro:

```ts
// @Filename: a.ts
interface Foo {}

// @Filename: b.ts
import { Foo } from './a';
```

### LINKED ISSUE #36614: `TypeError: Cannot read property 'kind' of undefined` in vscode build
Branch up at https://github.com/microsoft/vscode/tree/jackson/ts-build-error

When I run the typescript build, it crashes with the below error:
```
TypeError: Cannot read property 'kind' of undefined
    at getErrorSpanForNode (/Users/jkearl/Developer/vscode/node_modules/typescript/lib/typescript.js:13429:22)
    at createDiagnosticForNodeInSourceFile (/Users/jkearl/Developer/vscode/node_modules/typescript/lib/typescript.js:13389:20)
    at Object.createDiagnosticForNode (/Users/jkearl/Developer/vscode/node_modules/typescript/lib/typescript.js:13380:16)
    at reportNonExportedMember (/Users/jkearl/Developer/vscode/node_modules/typescript/lib/typescript.js:36179:50)
    at getExternalModuleMember (/Users/jkearl/Developer/vscode/node_modules/typescript/lib/typescript.js:36160:33)
    at getTargetOfImportSpecifier (/Users/jkearl/Developer/vscode/node_modules/typescript/lib/typescript.js:36187:20)
    at getTargetOfAliasDeclaration (/Users/jkearl/Developer/vscode/node_modules/typescript/lib/typescript.js:36238:28)
    at resolveAlias (/Users/jkearl/Developer/vscode/node_modules/typescript/lib/typescript.js:36277:30)
    at checkAliasSymbol (/Users/jkearl/Developer/vscode/node_modules/typescript/lib/typescript.js:64208:26)
    at checkImportBinding (/Users/jkearl/Developer/vscode/node_modules/typescript/lib/typescript.js:64240:13)
```

### PR DESCRIPTION
**Pull request by andrewbranch**
Fixes #36592
Fixes #36614

### CODE REVIEW

File: src/compiler/checker.ts [hunk 1]
```diff
@@ -2530,7 +2530,7 @@ namespace ts {
                     : error(name, Diagnostics.Module_0_declares_1_locally_but_it_is_not_exported, moduleName, declarationName);
 
                 addRelatedInfo(diagnostic,
-                    createDiagnosticForNode(localSymbol.valueDeclaration, Diagnostics._0_is_declared_here, declarationName)
+                    createDiagnosticForNode(localSymbol.valueDeclaration ?? localSymbol.declarations[0], Diagnostics._0_is_declared_here, declarationName)
```
Comment (andrewbranch) [inline src/compiler/checker.ts]: There could be more than one declaration, but exporting _any_ of them will fix the error, so 🤷‍♂

Comment (DanielRosenwasser) [reply src/compiler/checker.ts]: You could also report an error on all of them which might feel cleaner and work well too.

**Comment by andrewbranch** [comment]
👍 updated

**Comment by andrewbranch** [comment]
@typescript-bot cherry-pick this to release-3.8

**Comment by DanielRosenwasser** [comment]
I've heard the bot is having trouble because of octokit.

File: src/compiler/checker.ts (hunk 1 above)
Comment (DanielRosenwasser) [reply src/compiler/checker.ts]: I'd probably just avoid the `...map` and use `forEach` but whatever you'd prefer.

**Comment by DanielRosenwasser** [comment]
This should probably go in the RC since 2 people just reported errors on it in the VS Code codebase.

**Pull request merged**